### DISCUSSION TITLE
Using wrong store location in `pnpm run ...`

### DISCUSSION THREAD

**Discussion by Arctomachine** [pnpm/pnpm#6740] [Q&A]
I have pnpm 8.5.1 from corepack.
`pnpm store path` outputs `...\AppData\Local\pnpm\store\v3`, which exists
`pnpm run dev` fails with error "cannot find module `...\AppData\Roaming\npm\pnpm-global\5\node_modules\ts-node\dist\bin.js`"

How to make it look for packages in correct folder?
Looks like it only happens for this particular package, because ide reads all packages and another app from this monorepo runs without errors.

**Comment by KSXGitHub** [comment]
Can I see the content of your `package.json`?

**Reply by Arctomachine** [reply]
```js
"scripts": {
	"dev": "cross-env PAYLOAD_CONFIG_PATH=src/payload.config.ts nodemon",
	"build:payload": "cross-env PAYLOAD_CONFIG_PATH=src/payload.config.ts payload build",
	"build:server": "tsc",
	"build": "yarn copyfiles && yarn build:payload && yarn build:server",
	"serve": "cross-env PAYLOAD_CONFIG_PATH=dist/payload.config.js NODE_ENV=production node dist/server.js",
	"copyfiles": "copyfiles -u 1 \"src/**/*.{html,css,scss,ttf,woff,woff2,eot,svg,jpg,png}\" dist/",
	"generate:types": "cross-env PAYLOAD_CONFIG_PATH=src/payload.config.ts payload generate:types",
	"generate:graphQLSchema": "cross-env PAYLOAD_CONFIG_PATH=src/payload.config.ts payload generate:graphQLSchema"
},
"dependencies": {
	"dotenv": "^8.2.0",
	"express": "^4.17.1",
	"payload": "1.11.0",
	"zod": "^3.21.4"
},
"devDependencies": {
	"@types/express": "^4.17.9",
	"copyfiles": "^2.4.1",
	"cross-env": "^7.0.3",
	"nodemon": "^2.0.6",
	"ts-node": "^9.1.1"
}
```

**Reply by KSXGitHub** [reply]
Now I want to see `src/payload.config.ts`, and if this file depends on other files, I want to see them too.

But if there are too many files, can you create a minimal reproduction sample repo?

**Reply by Arctomachine** [reply]
Minimal reproduction:

1. `pnpm init`
2. `pnpm add ts-node`
3. `ts-node index.ts` (with or without actual index.ts file, as error happens before it gets to read this file)

**Reply by KSXGitHub** [reply]
So I opened a Windows VM on my PC, installed `pnpm` by `corepack enable`, and follow your steps. It doesn't reproduce the same behavior at what you describe:

![screenshot 1](https://github.com/pnpm/pnpm/assets/11488886/ec593056-4c67-4c6b-8445-a66ea3620f8b)
![screenshot 2](https://github.com/pnpm/pnpm/assets/11488886/3e910a13-7041-451e-a61e-47d7528978ec)

**Reply by Arctomachine** [reply]
Yes, obviously. Because it is normal behavior, and I am facing some kind of bug. https://github.com/pnpm/pnpm/issues/6751#issuecomment-1622281638 here I described what I tried to do to fix it, but it always keeps looking in that wrong store only for that one package, the rest work as expected.

**Reply by KSXGitHub** [reply]
Can you list all pnpm related environment variables?

If you have bash: `env | grep PNPM`
If you only have powershell: `dir env: | select-string PNPM`

**Reply by Arctomachine** [reply]
Bash (from git) only shows PNPM_HOME, which is AppData\Local\pnpm. Powershell shows nothing at all

**Reply by KSXGitHub** [reply]
Can you run `which -a pnpm` in bash?

**Reply by Arctomachine** [reply]
Program Files/nodejs/pnpm

**Reply by KSXGitHub** [reply]
So only one `pnpm`.

Can you run `pnpm exec which -a nodemon` in your project folder (in bash)? And if there is only one path, please run `pnpm exec bash -c 'cat "$(which nodemon)"'`.

**Reply by Arctomachine** [reply]
```
#!/bin/sh                                            
basedir=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")
                                                     
case `uname` in                                      
    *CYGWIN*) basedir=`cygpath -w "$basedir"`;;      
esac                                                 
                                                     
if [ -z "$NODE_PATH" ]; then                         
  export NODE_PATH="/mnt/c/Projects/pnpm-reproduction/node_modules/.pnpm/nodemon@2.0.6/node_modules/nodemon/bin/node_modules:/mnt/c/Projects/pnpm-reproduction/node_modules/.pnpm/nodemon@2.0.6/node_modules/nodemon/node_modules:/m
nt/c/Projects/pnpm-reproduction/node_modules/.pnpm/nodemon@2.0.6/node_modules:/mnt/c/Projects/pnpm-reproduction/node_modules/.pnpm/node_modules"
else
  export NODE_PATH="/mnt/c/Projects/pnpm-reproduction/node_modules/.pnpm/nodemon@2.0.6/node_modules/nodemon/bin/node_modules:/mnt/c/Projects/pnpm-reproduction/node_modules/.pnpm/nodemon@2.0.6/node_modules/nodemon/node_modules:/m
nt/c/Projects/pnpm-reproduction/node_modules/.pnpm/nodemon@2.0.6/node_modules:/mnt/c/Projects/pnpm-reproduction/node_modules/.pnpm/node_modules:$NODE_PATH"
fi
if [ -x "$basedir/node" ]; then
  exec "$basedir/node"  "$basedir/../nodemon/bin/nodemon.js" "$@"
else
  exec node  "$basedir/../nodemon/bin/nodemon.js" "$@"
fi
```

**Reply by Arctomachine** [reply]
And why nodemon when it is not the main cause of this problem?
